### PR TITLE
tcc: add missing archs, musl and make_check_target

### PR DESCRIPTION
**Pull request by m3tav3rse**

### CODE REVIEW

**Comment by maxice8** [comment]
bump revision= so it is rebuilt and the changes picked up.

any reason it might not work with i686-musl ?

**Comment by m3tav3rse** [comment]
Probably none but I haven't tested it on other archs

**Comment by Johnnynator** [comment]
@maxice8 Revision has to be bumped only when the resulting package will be changed. Since this PR effectively only adds one NEW architecture and doesn't touch the two existing one, it doesn't have to be revbumped.

**Comment by m3tav3rse** [comment]
What about arm and mips?
I would also add make_check_target but it segfaults on asm test.

**Comment by Johnnynator** [comment]
If they are official supported then drop the only_for_archs line and either get it to cross compile or set `nocross=yes`

File: srcpkgs/tcc/template [hunk 1]
```diff
@@ -15,6 +15,13 @@ license="LGPL-2.1"
 distfiles="http://repo.or.cz/tinycc.git/snapshot/${_gitrev}.tar.gz>${pkgname}-${version}.tar.gz"
 checksum=1c3a4150b4aaab9751af2112e8bfd8a93fd1a9d36af048e996ebfc0fd1f07e48
 
+case "$XBPS_TARGET_MACHINE" in
+	*-musl) configure_args="--config-musl";;
+esac
+case "$XBPS_TARGET_MACHINE" in
+	aarch64*|armv[567]*|i686*) nostrip=yes;;
```
Comment (Johnnynator) [inline srcpkgs/tcc/template]: no! This is only a workaround so xbps-src doesn't complain, but I highly doubt that it fixes cross builds.
`usr/bin/aarch64-linux-gnu-strip: Unable to recognise the format of the input file `/destdir/aarch64-linux-gnu/tcc-0.9.27/usr/bin/tcc'` <- this line normally indicates that the binary is not cross compiled/not in the target architecture.
The build-system would need to be patched to get it cross compiled or just set `nocross=yes`

Comment (m3tav3rse) [reply srcpkgs/tcc/template]: I'll go with nocross then

**Pull request merged**